### DISCUSSION TITLE
Dependency Dashboard on self-hosted GitLab is huge (>600k characters)

### DISCUSSION THREAD

**Discussion by jonasbadstuebner** [renovatebot/renovate#33044] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab + Renovate 39.62.1 through HelmChart

### Please tell us more about your question or problem

I copy-pasted the dependency dashboard from the edit-view of GitLab into a file and it is 615kb in size.
```sh
615K 11. Dez 17:04 dep-dashboard.md
```

It exceeds a size that allows for the checkboxes to render in time:
![grafik](https://github.com/user-attachments/assets/27e9ea7d-6a69-4996-b2c6-3d96a7f62c29)

The main content is the `Detected dependencies` section. Is there a way to disable this section only?
Or could you truncate the size of the dashboard?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Renovate generally truncates to the maximum size allowed by each platform. For modern GitLab that's 1m: https://github.com/renovatebot/renovate/blob/66dcb61908dfdbc262c0d61861145a6e8240f312/lib/modules/platform/gitlab/index.ts#L890-L900

A few possible solutions here:
- Allow custom truncation length e.g. as an experimental env
- Reduce the truncation length for GitLab for all, if it's impractical
- Allow full templating of dependency dashboard so that this detected dependencies section can be removed via config

**Reply by jonasbadstuebner** [reply]
I think option 2 is rather a workaround than a real solution. Because if I have thousands of pending merge requests, maybe I want to see them all?

So I would prefer option 3, as it would fix my current problem, in combination with option 1 to reduce the limit for the future.

**Reply by rarkins** [reply]
If you're interested in submitting a PR for this, feel free to send me an email to join our contributor Slack

**Reply by jonasbadstuebner** [reply]
I sent you a mail, I hope I can create a PR for that soon.
